garden-backend-service/src/config.py

```diff
@@ -0,0 +1,27 @@
+from functools import lru_cache
+
+from pydantic import BaseSettings
+from dotenv import find_dotenv
+
+
+class Settings(BaseSettings):
+    DEBUG: bool = False
+    LOGLEVEL: str = "INFO"
+    API_CLIENT_ID: str = "API_CLIENT_ID"
+    API_CLIENT_SECRET: str = "API_CLIENT_SECRET"
+
+    GARDEN_DEFAULT_SCOPE = (
+        "https://auth.globus.org/scopes/0948a6b0-a622-4078-b0a4-bfd6d77d65cf/action_all"
+    )
+
+    class Config:
+        case_sensitive = True
+        env_file = find_dotenv()
+
+
+@lru_cache()
+def get_settings() -> Settings:
+    return Settings()
```

Comment:
Understanding check: the `find_dotenv` call is the thing worth caching here?

Reply:
I was maybe a little cargo-culty in plagiarizing this from globus compute because its purpose wasn't 100% obvious to me either, but my take on it is that it's probably a good idea for everything to share the same `Settings()` instance instead of instantiating their own. the full-stack-fastapi-template repo I was also looking at [does something similar](https://github.com/tiangolo/full-stack-fastapi-template/blob/c947cb09e1d03084c5037e2ef69d58c1c0397960/backend/app/core/config.py#L121) with a top level variable it reuses everywhere instead of instantiating `Settings()` more than once. 

the pydantic base class we use here automatically reads environment variables (set by dotenv or otherwise), and my suspicion is that they got burned by some subtle interaction between the state of os.environ and the pydantic class here, but I didn't dig too deeply